### PR TITLE
BL 11952 Update lambda event trigger to be a SQS Event

### PR DESCRIPTION
**Pull request by davidmck95**
Includes logic changes as the hvt-record-handler lambda has already checked if availability has changed before publishing to email SNS topic, which means that we don't need to check this here anymore.

### CODE REVIEW

File: src/lib/email.ts [hunk 1]
```diff
@@ -14,8 +14,6 @@ export const buildEmailBody = (params: BuildEmailBodyParams): string => {
   const linkTemplateTag = availability.isAvailable
     ? 'link'
     : 'link';
-  const tokenKey = availability.isAvailable ? 'no' : 'yes';
```
Comment (davidmck95) [inline src/lib/email.ts:17]: this was not being used so I removed

**Review by sdh100shaun — APPROVED** [review]

File: src/handler/index.ts [hunk 2]
```diff
@@ -35,45 +35,43 @@ export const handler = async (event: DynamoDBStreamEvent, context: Context): Pro
...
       return;
     }
 
-    logger.debug(`Processing record: ${JSON.stringify(record)}`);
-
     // Extract data from event
     let availabilityData: AvailabilityChangeData;
     try {
-      availabilityData = extractAvailabilityData(record);
+      availabilityData = validateAvailabilityData(message);
```
Comment (sdh100shaun) [inline src/handler/index.ts:44]: As per discussion - shall we change this ? and consolidate the error handling

File: src/handler/index.ts [hunk 3]
```diff
@@ -35,45 +35,43 @@ export const handler = async (event: DynamoDBStreamEvent, context: Context): Pro
...
     } catch (err2) {
       logger.error((err2 as Error).message);
       return; // skip to next record
     }
-    const {
-      oldAvailability,
-      newAvailability,
-    } = availabilityData;
+    const { availability } = availabilityData;
```
Comment (sdh100shaun) [inline src/handler/index.ts]: should we expand the destructuring?

**Review by sdh100shaun — CHANGES_REQUESTED** [review]

File: src/handler/index.ts (hunk 2 above)
Comment (AnnaDodson) [reply src/handler/index.ts:44]: Consolidated the try catch and handled the different errors by checking the type of error thrown

File: src/handler/index.ts (hunk 3 above)
Comment (AnnaDodson) [reply src/handler/index.ts]: I've got rid of the destrucing on availability as none of the others were getting deconstructed just to be set again into the object for `buildSQSMessage` to handle

**Review by sdh100shaun — DISMISSED** [review]

**Review by sdh100shaun — APPROVED** [review]

**Review by AnnaDodson — APPROVED** [review]

**Pull request merged**